Docs (#2528)
TuSimple/naive-ui: demo/pages/docs/import-on-demand/enUS/index.md

For more info about theming, see [Customizing Theme](customize-theme).

## On-demand Import(Auto)

If you develop using the template approach, you can use the `unplugin-vue-components` plugin to automatically load components on demand.

The plugin will automatically parse the components used in the template and import the components.

> ```vite.config.js
> import { defineConfig } from 'vite'

Reviewer: ```suggestion
```ts
// vite.config.ts
import { defineConfig } from 'vite'

```

import vue from '@vitejs/plugin-vue'
import Components from 'unplugin-vue-components/vite';
import { NaiveUiResolver } from 'unplugin-vue-components/resolvers';

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [vue(),
    Components({
      resolvers: [
        NaiveUiResolver()
      ]
    })
  ]
});
```

> Note: This method does not handle function API components, such as the `useMessage` component, and the user still needs to import the corresponding component manually.

Reviewer: ```suggestion
Note: This method does not work with composables such as `useMessage`, and the user still needs to import the corresponding composable manually.
```

</style>
```

> ## Install on Demand Globally(Manual)

Reviewer: ```suggestion
## Install on Demand Globally (Manually)
```
